modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/configuration/CETConfigurationFactory.java

```diff
@@ -74,25 +108,112 @@ protected void modified(Map<String, Object> properties) throws Exception {
 				_cet = _cetManager.addCET(
 					cetConfiguration, companyId,
 					_getExternalReferenceCode(properties));
+
+				if (Objects.equals(
+						_cet.getType(),
+						ClientExtensionEntryConstants.TYPE_THEME_CSS)) {
+
+					_clientExtensionEntryRelLocalService.
+						deleteClientExtensionEntryRels(
+							companyId, _cet.getExternalReferenceCode());
+
+					_addControlPanelThemeCSSClientExtensionEntryRel(companyId);
+				}
 			});
 	}
 
+	private void _addControlPanelThemeCSSClientExtensionEntryRel(Long companyId)
+		throws PortalException {
+
+		ThemeCSSCET themeCSSCET = (ThemeCSSCET)_cet;
+
+		if (!Objects.equals(themeCSSCET.getScope(), "controlPanel")) {
+			return;
+		}
+
+		ClientExtensionEntryRel clientExtensionEntryRel =
+			_clientExtensionEntryRelLocalService.
+				fetchClientExtensionEntryRelByExternalReferenceCode(
+					themeCSSCET.getExternalReferenceCode(), companyId);
+
+		if (clientExtensionEntryRel != null) {
+			return;
+		}
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		User guestUser = company.getGuestUser();
+
+		Layout layout = _getControlPanelLayout(companyId);
+
+		_clientExtensionEntryRelLocalService.addClientExtensionEntryRel(
+			guestUser.getUserId(), layout.getGroupId(),
+			_portal.getClassNameId(Layout.class), layout.getPlid(),
+			_cet.getExternalReferenceCode(), _cet.getType(), StringPool.BLANK,
+			new ServiceContext());
+
+		int themeCSSClientExtensionEntryRelsCount =
```

Comment:
This is much better but, just a little bit more... Following the name of the method this is in and being more specific lets go for this:

controlPanelThemeCSSClientExtensionEntryRelsCount

Reply:
Since the method name already set the context for "control panel theme css",  why not go with "`clientExtensionEntryRelsCount`" only ? It's already implicit that is related with control panel theme css

Reply:
@ethib137 I agree with @algobob. We are overthinking this

Reply:
Okay, I'll trust you guys.